### PR TITLE
feat(print): support touch drag for print extent drawing

### PR DESCRIPTION
**Pull request by Varshavia**
Adds single-finger touch support to the "Draw print extent" tool in the Print Layout dialog, which was previously mouse-only (a TODO in the code called this out as a follow-up). A second finger joining mid-drag cancels the draw rather than committing an extent from an ambiguous multi-touch point, and touchZoomRotate/touchPitch are suspended for the duration of the draw and restored after, matching the existing dragPan/scrollZoom/doubleClickZoom handling.

Also adds tests/print-extent.test.ts (9 tests) — this module had no prior coverage.

Verified against the actual maplibre-gl@5.24.0 type definitions (MapTouchEvent.point, touchZoomRotate, touchPitch). Manual QA on a real touchscreen/tablet is recommended before merge — the test suite covers the interaction logic in a headless harness, not real touch input.

## Summary by CodeRabbit

* **New Features**
  * Added touch support for drawing print extents, including single-finger dragging and two-finger gesture handling.
  * Temporarily pauses map touch zoom/rotation and pitch interactions during drawing, then restores them afterward.
* **Bug Fixes**
  * Improved cancellation and edge-case handling for Escape, multi-touch entry, touchcancel, and interrupted drags to prevent incorrect commits.
* **Tests**
  * Added DOM-driven mouse/touch test coverage, including deterministic interaction simulations and verification that handlers are correctly restored.

### CODE REVIEW

File: apps/geolibre-desktop/src/lib/print-extent.ts [hunk 1]
```diff
@@ -308,15 +331,55 @@ export function drawPrintExtent(
...
+        return;
+      }
+      queueCanvasMove({ x: e.point.x, y: e.point.y }, touchShiftKey(e));
+    };
+    const onTouchEnd = (e: MapTouchEvent) => {
+      commit({ x: e.point.x, y: e.point.y }, touchShiftKey(e));
+    };
+    // The OS or browser reclaiming the gesture (e.g. an incoming call, a
+    // system edge-swipe) — there is no completed drag to commit.
+    const onTouchCancel = () => finish(null);
```
[the remark this answers is not in the document]
Comment (giswqs) [reply apps/geolibre-desktop/src/lib/print-extent.ts:392]: Done in 9ea467cd — `onTouchStart` now calls `finish(null)` when `start` is already set, and `tests/print-extent.test.ts` gained "cancels the draw when a second finger lands mid-draw without moving" (single-finger touch → two-point touchstart → asserts null). Confirmed the mechanism against the vendored maplibre-gl 5.24.0 rather than docs: `maplibre-gl-dev.js:66094` builds points from `changedTouches` only for `touchend` and from `touches` otherwise, so a second finger does re-fire `touchstart` with both contacts. The new test fails (hangs, promise never settles) without the guard.

File: apps/geolibre-desktop/src/lib/print-extent.ts [hunk 2]
```diff
@@ -308,15 +331,55 @@ export function drawPrintExtent(
...
+      // multi-touch center point, and let the user restart the tool.
+      if (e.points.length > 1) {
+        finish(null);
+        return;
+      }
+      queueCanvasMove({ x: e.point.x, y: e.point.y }, touchShiftKey(e));
+    };
+    const onTouchEnd = (e: MapTouchEvent) => {
+      commit({ x: e.point.x, y: e.point.y }, touchShiftKey(e));
+    };
```
[the remark this answers is not in the document]
Comment (giswqs) [reply apps/geolibre-desktop/src/lib/print-extent.ts:389]: Partly done in 9ea467cd. Verified your `changedTouches` premise against the vendored maplibre-gl 5.24.0 (not just docs) — `maplibre-gl-dev.js:66094`: `const touches = type === 'touchend' ? originalEvent.changedTouches : originalEvent.touches;`. Correct on both counts.

**Took the `onTouchStart` half.** It now cancels when a draw is already armed, as you suggested. Added two regressions, both of which fail without it — including the exact narrow ordering you called out (second finger down-then-up, no intervening `touchmove`), which previously committed a bogus `[6, 6, 46, 46]` from the wrong finger.

**Declined the `onTouchEnd` guard.** `if (e.originalEvent.touches.length > 0) return;` would introduce a worse bug than it fixes: `touches` is *surface-wide*, not canvas-scoped, so a contact that never touched the map at all — a thumb steadying a tablet, a finger resting on a sidebar — sits in that list and would silently swallow a legitimate single-finger release. `commit()` returns early, nothing resolves, and the tool stays armed indefinitely (only Escape/blur recovers). That is a hang, versus the narrow mis-commit it guards.

It is also unnecessary once `onTouchStart` cancels. Touch events only route to the element the touch *started* on, so a canvas `touchend`'s `changedTouches` can only contain canvas-originated fingers — and any second canvas-originated finger now cancels the draw at `touchstart`. So while `start` is set, exactly one such finger is down and it is the one lifting, making `e.point` (the `changedTouches` centroid) already the right point.

Added "commits a single-finger drag even while an unrelated contact stays on the surface" to pin that down, and a comment at the handler so the omission reads as deliberate. Happy to revisit if you see a path where a non-canvas-originated touch reaches `changedTouches` here.
